### DISCUSSION TITLE
upgrade-interactive with workspaces

### DISCUSSION THREAD

**Discussion by stefanomiccoli** [yarnpkg/berry#2567] [Q&A]
when using yarn 1, upgrade-interactive was very useful to manage upgrade of dependencies for each workspace, because together with the dependency name and current and latest version, also the workspace the dependency belong to was shown.

Now, after migrating to 2, if i run `yarn upgrade-interactive` (after installing `yarn plugin import interactive-tools`) the dependencies are listed without any reference to the workspace they belong to.

Needless to say, if 2 packages have the same dependency it's basically impossible to get which workspace is going to be updated.

e.g.
```
@types/aws-lambda                             ◉ ^8.10.17       ◯ ^8.10.72      
@types/aws-lambda                             ◉ ^8.10.70       ◯ ^8.10.72      
@types/aws-lambda                             ◉ ^8.10.71       ◯ ^8.10.72      
```

Am I missing something? is Yarn2 missing something?

thanks for the help

**Comment by stefanomiccoli** [comment]
besides, I then rolledback to yarn 1, run the command again for comparison. a completely different set of packages to be updated.

```
devDependencies
   name                 range   from         to        workspace               url
 ( ) @types/express       latest  4.17.8    ❯  4.17.11   @<my-scope>/<my-package>        https://github.com/DefinitelyTyped/DefinitelyTyped.git
```

**Reply by JackCA** [reply]
Agreed. It's difficult to figure out where the package lives without a workspace column

**Comment by smiccoli** [comment]
any news?
